app/controllers/course/stage/instructions.ts

```diff
@@ -47,7 +50,13 @@ export default class CourseStageInstructionsController extends Controller {
   }
 
   get shouldShowLanguageGuide() {
-    return !this.model.courseStage.isFirst && (this.featureFlags.canSeeLanguageGuidesForStage2 || this.authenticator.currentUser?.isStaff);
+    const languageGuide = this.model.courseStage.languageGuides.findBy('language', this.model.activeRepository.language);
+
+    return (
+      !this.model.courseStage.isFirst &&
+      !!languageGuide &&
+      (this.featureFlags.canSeeLanguageGuidesForStage2 || this.authenticator.currentUser?.isStaff)
```

Comment:
```suggestion
      this.featureFlags.canSeeLanguageGuidesForStage2
```

(We already have the staff check inside `canSeeLanguageGuidesForStage2`)